fluent-bit-kinesis.go

```diff
@@ -118,21 +119,49 @@ func FLBPluginInit(ctx unsafe.Pointer) int {
 //export FLBPluginFlushCtx
 func FLBPluginFlushCtx(ctx, data unsafe.Pointer, length C.int, tag *C.char) int {
 	kinesisOutput := getPluginInstance(ctx)
+
+	curRetries := kinesisOutput.GetConcurrentRetries()
+	if curRetries > concurrentRetryLimit {
+		logrus.Infof("[kinesis] flush returning retry, too many concurrent retries (%d)\n", curRetries)
+		return output.FLB_RETRY
+	}
+
 	events, count, retCode := unpackRecords(kinesisOutput, data, length)
 	if retCode != output.FLB_OK {
+		logrus.Errorf("[kinesis] failed to unpackRecords\n")
 		return retCode
 	}
 	go flushWithRetries(kinesisOutput, tag, count, events, retries)
 	return output.FLB_OK
 }
 
 func flushWithRetries(kinesisOutput *kinesis.OutputPlugin, tag *C.char, count int, records []*kinesisAPI.PutRecordsRequestEntry, retries int) {
-	for i := 0; i < retries; i++ {
-		// TODO: Would probably want to backoff before retrying?
-		retCode := pluginConcurrentFlush(kinesisOutput, tag, count, records)
+	var retCode, tries int
+
+	backoff := kinesisOutput.GetConcurrentRetries()
```

Comment:
I think the name is misleading- calling it something like currentRetries would be better

Reply:
Noted... update coming